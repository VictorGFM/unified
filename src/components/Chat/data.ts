import { Message } from '../Message/Message'

export const messages: Message[] = [
  { text: 'Some user message...', author: 'Victor Moraes' },
  { text: 'Some contact message...', author: 'Contact Name' },
]
