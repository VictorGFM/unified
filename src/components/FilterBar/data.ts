import { FiltersGroupsData } from './FilterBar'

export const Filters: FiltersGroupsData = {
  Posts: [
    {
      id: '1',
      label: 'Instagram Posts',
    },
    {
      id: '2',
      label: 'Twitter Posts',
    },
    {
      id: '3',
      label: 'Facebook Posts',
    },
  ],
  Groups: [
    {
      id: '4',
      label: 'Grupo dos Amigos',
    },
    {
      id: '5',
      label: 'Grupo do Trabalho',
    },
    {
      id: '6',
      label: 'Grupo da Família',
    },
    {
      id: '7',
      label: 'Grupo dos Canários',
    },
  ],
  Others: [
    {
      id: '8',
      label: 'Other Filter',
    },
    {
      id: '9',
      label: 'Other Filter',
    },
    {
      id: '10',
      label: 'Other Filter',
    },
  ],
}
