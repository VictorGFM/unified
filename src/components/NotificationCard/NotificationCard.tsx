import { SocialType } from '../SocialMediaBar/SocialMediaBar'
import {
  ContactName,
  ContactPhoto,
  NotificationCardDiv,
  NotificationDescription,
  NotificationSocialTag,
  NotificationTime,
} from './styles'

export interface Notification {
  id: string
  contactName: string
  contactPhoto: string
  description: string
  time: string
  socialType: SocialType
}

interface NotificationProps {
  notification: Notification
  onClick: () => void
}

const NotificationCard = ({ notification, onClick }: NotificationProps) => {
  const { contactName, contactPhoto, description, time, socialType } = notification
  return (
    <NotificationCardDiv onClick={onClick}>
      <ContactPhoto src={contactPhoto} />
      <ContactName>{contactName}</ContactName>
      <NotificationDescription>{description}</NotificationDescription>
      <NotificationTime>{time}</NotificationTime>
      <NotificationSocialTag socialType={socialType} />
    </NotificationCardDiv>
  )
}

export default NotificationCard
