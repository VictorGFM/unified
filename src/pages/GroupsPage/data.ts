import { GroupImage1, GroupImage2, GroupImage3, GroupImage4 } from '../../assets'
import { Group } from '../../components/GroupCard/GroupCard'

export const groups: Group[] = [
  { id: '1', title: 'Grupo dos Amigos', image: GroupImage1 },
  { id: '2', title: 'Grupo da Família', image: GroupImage2 },
  { id: '3', title: 'Grupo do Trabalho', image: GroupImage3 },
  { id: '4', title: 'Grupo dos Canários', image: GroupImage4 },
]
