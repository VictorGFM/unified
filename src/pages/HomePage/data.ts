import { NumberInteractions, PostInterface } from '../../components/Post/Post'
import { AlfredoPhotoIcon, AndromedaImage, ContactPhotoIcon, ProfilePhotoIcon } from '../../assets'

export const interactions: NumberInteractions[] = [
  {
    likes: 237,
    comments: 15,
    shares: 10,
  },
  {
    likes: 107,
    comments: 10,
    shares: 7,
  },
  {
    likes: 5,
    comments: 2,
    shares: 1,
  },
  {
    likes: 527,
    comments: 132,
    shares: 97,
  },
]

export const posts: PostInterface[] = [
  {
    id: '1',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '1h',
    socialType: 'Instagram',
    postDescription: 'Look at this galaxy! Its so beautiful...',
    postImage: AndromedaImage,
    numberInteractions: interactions[0],
  },
  {
    id: '2',
    contactName: 'Victor Moraes',
    contactPhoto: ProfilePhotoIcon,
    contactGroup: 'Grupo da Família',
    postTime: '3h',
    socialType: 'Twitter',
    postDescription:
      'So, this is a description of a Twitter post, but right after an Instagram post. All in the same place, awesome,' +
      ' isnt it?',
    numberInteractions: interactions[1],
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    contactGroup: 'Grupo dos Canários',
    postTime: '15h',
    socialType: 'Facebook',
    postDescription: 'Some post description...',
    numberInteractions: interactions[2],
  },
  {
    id: '4',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    contactGroup: 'Grupo dos Amigos',
    postTime: '21h',
    socialType: 'Twitter',
    postDescription: 'Au Au Au...',
    numberInteractions: interactions[3],
  },
]
