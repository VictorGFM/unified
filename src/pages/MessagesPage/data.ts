import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { ContactInfo } from '../../components/ContactCard/ContactCard'
import dayjs from 'dayjs'

export const defaultMessages = {
  instagram: [
    {
      date: dayjs('2021-09-02'),
      messages: [
        {
          text: 'My message...',
          author: 'Victor Moraes',
          datetime: dayjs('2021-09-02T13:05:00Z'),
        },
      ],
    },
    {
      date: dayjs('2021-09-03'),
      messages: [
        {
          text: 'Some message...',
          author: 'Contact Name',
          datetime: dayjs('2021-09-03T14:10:00Z'),
        },
      ],
    },
  ],
  twitter: [],
  facebook: [],
}

export const contacts: ContactInfo[] = [
  {
    id: '1',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    status: 'online',
    primaryInfo: 'Some message...',
    secondaryInfo: 'Dec 15, 2021',
    messages: {
      instagram: [
        {
          date: dayjs('2021-09-05'),
          messages: [
            {
              text: 'My message...',
              author: 'Victor Moraes',
              datetime: dayjs('2021-09-05T22:10:00Z'),
            },
          ],
        },
        {
          date: dayjs('2021-12-15'),
          messages: [
            {
              text: 'Some message...',
              author: 'Contact Name',
              datetime: dayjs('2021-12-15T18:20:00Z'),
            },
          ],
        },
      ],
      twitter: [],
      facebook: [],
    },
  },
  {
    id: '2',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    status: 'online',
    primaryInfo: 'Au Au Au...',
    secondaryInfo: 'Dec 5, 2021',
    contactNotifications: {
      total: 5,
      messages: {
        total: 5,
        instagram: 1,
        twitter: 3,
        facebook: 1,
      },
    },
    messages: {
      instagram: [
        {
          date: dayjs('2021-09-05'),
          messages: [
            {
              text: 'My message...',
              author: 'Victor Moraes',
              datetime: dayjs('2021-09-05T22:10:00Z'),
            },
          ],
        },
        {
          date: dayjs('2021-12-05'),
          messages: [
            {
              text: '( ._.)',
              author: 'Alfredo',
              datetime: dayjs('2021-12-05T18:20:00Z'),
            },
          ],
        },
      ],
      twitter: [
        {
          date: dayjs('2021-09-07'),
          messages: [
            {
              text: '( ._.)',
              author: 'Alfredo',
              datetime: dayjs('2021-09-07T18:20:00Z'),
            },
          ],
        },
        {
          date: dayjs('2021-09-15'),
          messages: [
            {
              text: '( ._.)',
              author: 'Alfredo',
              datetime: dayjs('2021-09-15T20:20:00Z'),
            },
            {
              text: '( ._.)',
              author: 'Alfredo',
              datetime: dayjs('2021-09-15T20:30:00Z'),
            },
          ],
        },
      ],
      facebook: [
        {
          date: dayjs('2021-12-05'),
          messages: [
            {
              text: 'Hi Alfredo!',
              author: 'Victor Moraes',
              datetime: dayjs('2021-12-05T21:45:00Z'),
            },
            {
              text: 'Au Au Au...',
              author: 'Alfredo',
              datetime: dayjs('2021-12-05T22:07:00Z'),
            },
          ],
        },
      ],
    },
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    contactNotifications: {
      total: 41,
      messages: {
        total: 31,
        instagram: 17,
        twitter: 11,
        facebook: 3,
      },
      stories: {
        total: 5,
        instagram: 3,
        facebook: 2,
      },
    },
    messages: {
      instagram: [
        {
          date: dayjs('2021-09-02'),
          messages: [
            {
              text: 'My message...',
              author: 'Victor Moraes',
              datetime: dayjs('2021-09-02T13:05:00Z'),
            },
          ],
        },
        {
          date: dayjs('2021-09-03'),
          messages: [
            {
              text: 'Some message...',
              author: 'Contact Name',
              datetime: dayjs('2021-09-03T14:10:00Z'),
            },
          ],
        },
      ],
      twitter: [
        {
          date: dayjs('2021-09-02'),
          messages: [
            {
              text: 'My message...',
              author: 'Victor Moraes',
              datetime: dayjs('2021-09-02T13:05:00Z'),
            },
          ],
        },
      ],
      facebook: [
        {
          date: dayjs('2021-09-03'),
          messages: [
            {
              text: 'Some message...',
              author: 'Contact Name',
              datetime: dayjs('2021-09-03T14:10:00Z'),
            },
          ],
        },
      ],
    },
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '6',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '7',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '8',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '9',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '10',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '11',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '12',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '13',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '14',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '15',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '16',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '17',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '18',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '19',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '20',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '21',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '22',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '23',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
  {
    id: '24',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    primaryInfo: 'Some message...',
    secondaryInfo: 'Sep 3, 2021',
    messages: defaultMessages,
  },
]
