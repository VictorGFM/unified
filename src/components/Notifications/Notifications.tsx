import { useEffect, useRef, useState } from 'react'
import { AppIcon, FacebookIcon, InstagramIcon, TwitterIcon } from '../../assets'
import NotificationCard from '../NotificationCard'
import { SocialType } from '../SocialMediaBar/SocialMediaBar'
import {
  NotificationsDiv,
  NotificationsHeader,
  NotificationsList,
  NotificationsSocialButton,
  NotificationsTitle,
  SocialButtonsDiv,
} from './styles'
import { notifications } from './data'
import { useNavigate } from 'react-router-dom'

interface Notifications {
  handleCloseNotifications: () => void
}

const Notifications = ({ handleCloseNotifications }: Notifications) => {
  const [selectedSocial, setSelectedSocial] = useState('Unified')
  const ref = useRef(null)
  const navigate = useNavigate()

  useEffect(() => {
    const handleClickOutside = event => {
      if (ref.current && !ref.current.contains(event.target)) {
        handleCloseNotifications()
      }
    }
    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [ref])

  const handleClickButton = (socialType: SocialType) => {
    setSelectedSocial(socialType)
  }

  const handleClickNotification = (socialType: SocialType) => {
    navigate(`/social/${socialType.toLowerCase()}`)
  }

  return (
    <NotificationsDiv ref={ref}>
      <NotificationsHeader>
        <NotificationsTitle>Notifications</NotificationsTitle>
        <SocialButtonsDiv>
          <NotificationsSocialButton
            icon={AppIcon}
            isSelected={selectedSocial === 'Unified'}
            onClick={() => handleClickButton('Unified')}
            unreadMessages={64}
          />
          <NotificationsSocialButton
            icon={InstagramIcon}
            isSelected={selectedSocial === 'Instagram'}
            onClick={() => handleClickButton('Instagram')}
            unreadMessages={2}
          />
          <NotificationsSocialButton
            icon={TwitterIcon}
            isSelected={selectedSocial === 'Twitter'}
            onClick={() => handleClickButton('Twitter')}
            unreadMessages={31}
          />
          <NotificationsSocialButton
            icon={FacebookIcon}
            isSelected={selectedSocial === 'Facebook'}
            onClick={() => handleClickButton('Facebook')}
            unreadMessages={31}
          />
        </SocialButtonsDiv>
      </NotificationsHeader>
      <NotificationsList>
        {notifications &&
          notifications.map(notification => (
            <NotificationCard
              key={notification.id}
              notification={notification}
              onClick={() => handleClickNotification(notification.socialType)}
            />
          ))}
      </NotificationsList>
    </NotificationsDiv>
  )
}

export default Notifications
