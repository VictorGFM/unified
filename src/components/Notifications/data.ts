import { AlfredoPhotoIcon, ContactPhotoIcon } from '../../assets'
import { Notification } from '../NotificationCard/NotificationCard'

export const notifications: Notification[] = [
  {
    id: '1',
    contactName: 'Alfredo',
    contactPhoto: AlfredoPhotoIcon,
    description: 'liked your tweet.',
    time: '1h',
    socialType: 'Twitter',
  },
  {
    id: '2',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '3',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'shared your post.',
    time: '1h',
    socialType: 'Facebook',
  },
  {
    id: '4',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '1h',
    socialType: 'Instagram',
  },
  {
    id: '5',
    contactName: 'Contact Name',
    contactPhoto: ContactPhotoIcon,
    description: 'liked your post.',
    time: '2h',
    socialType: 'Twitter',
  },
]
