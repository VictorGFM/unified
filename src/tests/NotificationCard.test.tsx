import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import NotificationCard from '../components/NotificationCard'
import { notifications } from '../components/Notifications/data'

describe('<NotificationCard/>', () => {
  it('should show match snapshot', () => {
    const component = render(
      <NotificationCard notification={notifications[0]} onClick={() => {}} />
    )

    expect(component).toMatchSnapshot()
  })
})
