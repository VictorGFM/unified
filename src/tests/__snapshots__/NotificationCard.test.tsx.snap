// Vitest Snapshot v1

exports[`<NotificationCard/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkzDqf hXIkkB"
      >
        <img
          class="sc-hKMtZM eaHsLj"
          src="/src/assets/icons/AlfredoPhotoIcon.svg"
        />
        <span
          class="sc-eCYdqJ jMxTVP"
        >
          Alfredo
        </span>
        <span
          class="sc-jSMfEi jCXEyz"
        >
          liked your tweet.
        </span>
        <span
          class="sc-gKXOVf fVYPhJ"
        >
          1h
        </span>
        <div
          class="sc-bczRLJ gpaXLH"
        >
          <img
            class="sc-gsnTZi gNNCuJ"
            src="/src/assets/icons/TwitterIcon.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkzDqf hXIkkB"
    >
      <img
        class="sc-hKMtZM eaHsLj"
        src="/src/assets/icons/AlfredoPhotoIcon.svg"
      />
      <span
        class="sc-eCYdqJ jMxTVP"
      >
        Alfredo
      </span>
      <span
        class="sc-jSMfEi jCXEyz"
      >
        liked your tweet.
      </span>
      <span
        class="sc-gKXOVf fVYPhJ"
      >
        1h
      </span>
      <div
        class="sc-bczRLJ gpaXLH"
      >
        <img
          class="sc-gsnTZi gNNCuJ"
          src="/src/assets/icons/TwitterIcon.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
